core/src/main/java/org/apache/metamodel/query/SelectItem.java

```diff
@@ -484,11 +498,24 @@ public StringBuilder toStringNoAlias(boolean includeSchemaInColumnPath) {
             final Object[] functionParameters = getFunctionParameters();
             if (functionParameters != null && functionParameters.length != 0) {
                 for (int i = 0; i < functionParameters.length; i++) {
-                    functionBeginning.append('\'');
-                    functionBeginning.append(functionParameters[i]);
-                    functionBeginning.append('\'');
+                    if (functionParameters[i] instanceof Column) {
+                        Column col = (Column) functionParameters[i];
+                        functionBeginning.append(getToStringColumnPrefix(includeSchemaInColumnPath, col));
+                        functionBeginning.append(col.getQuotedName());
+                    } else {
+                        if (!_function.getFunctionName().equals(FunctionType.CONCAT.getFunctionName())) {
+                            functionBeginning.append('\'');
+                        }
+                        functionBeginning.append(functionParameters[i]);
+                        if (!_function.getFunctionName().equals(FunctionType.CONCAT.getFunctionName())) {
+                            functionBeginning.append('\'');
+                        }
```

Comment:
Also the changes here seem very unfortunate because they have a specific-case exception for something that is very general. If we instead just made sure that _any_ function could have select item as parameter, then we're in a much better position.